rfd/0077-support-dump.md

```diff
@@ -0,0 +1,252 @@
+---
+authors: Jeff Anderson <jeff@goteleport.com>
+state: draft
+---
+
+# RFD 77 - Support Dump
+
+## What
+
+A support dump process should be able to gather as much relevant information
+about a given teleport instance as possible. This information should be
+programatically discovered and package it up to be shared easily.
+
+## Why
+
+It is necessary to collect a variety of basic information when assisting a
+teleport administrator with an issue remotely. This information gathering step
+can be high effort for the user requesting help, especially since they may not
+be able to anticipate which data is relevant.
+
+This will primarily assist the interactions between the Teleport support team
+and customers requesting help. The Teleport development team should also
+greatly benefit from a standardized set of information about a problem cluster.
+
+Without a support dump process, this increases the burden on both ends of the
+interaction as relevant information is requested and inspected. This adds
+unnecessary time to solution with each subsequent requests for information. 
+
+A standardized support dump would be able to provide a large amount of
+information in one step, increasing visibility into the possible issues and
+decreasing time to solution for a given request. It will also greatly decrease
+the effort required to reproduce a particular issue, especially when there is a
+less common configuration in play.
+
+## Details
+
+### End User Experience
+
+Because teleport has many deployment methods, the support dump procedure should
+be smart enough to detect and gather information in the correct way. This may
+include an entire cluster, or a single agent. It should also be able to gather
+logs from standard locations wherever possible. It should advise the end user
```

Comment:
Logs strike me as something that we'll need to explore, as they handled at a platform level and not in Teleport itself.

Reply:
yes indeed-- these are often critical for troubleshooting. The support dump processes I've seen in the past will check 2-5 common places where the log may be, and then give up with a warning. This is fine as long as the warning is not silent.

I have done some experiments with trying to detect if teleport is running in systemd or in a container and then trying to see if it has access to the respective system. The mvp for this RFD will be to only check for logs immediately available to teleport. That would be places like journald or the file as configured in the teleport.yaml config.

I consider more specialized cases like getting logs from kubernetes are out of scope for this particular proposal.